packages/react-scripts/scripts/start.js

```diff
@@ -11,6 +11,11 @@
 
 process.env.NODE_ENV = 'development';
 
+// Load environment variables from .env file. Surpress warnings using silent
+// if this file is missing. dotenv will never modify any environment variables
+// that have already been set.
+require('dotenv').config({silent: true});
```

Comment:
I think there are use cases for this PR outside the `REACT_APP_*` usecases eg. [absolute imports](https://github.com/facebookincubator/create-react-app/pull/693). 

By setting `NODE_PATH=.` I can start using absolute imports in my codebase:

Relative:

```js
import Button from '../../shared/button';
``` 
vs. 

```js
import Button from 'src/shared/button';
```